dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,42 +53,153 @@ public Pattern getParameterPattern() {
+    }
+
+    DatabaseObjectName buildDatabaseObjectName(String objName) throws SQLException {
+        String[] qualifiers = objName.toUpperCase().split("\\.");
+        String schemaName = (qualifiers.length == 2) ? qualifiers[0] : getConnection().getSchema();
+        String objectName = (qualifiers.length == 2) ? qualifiers[1] : qualifiers[0];
+        return new DatabaseObjectName(schemaName, objectName);
     }
 
+    @Override
+    public Map<String, DbParameterAccessor> getAllProcedureParameters(String procName)
+            throws SQLException {
+        DatabaseObjectName name = buildDatabaseObjectName(procName);
+        // https://github.com/apache/derby/blob/10.13/java/engine/org/apache/derby/impl/jdbc/metadata.properties
+        String sql =
+        "SELECT V.COLUMN_NAME" +
+        "           AS COLUMN_NAME" +
+        "     , CASE WHEN A.ALIASTYPE = 'P'" +
+        "            THEN CASE WHEN V.COLUMN_TYPE = " + DatabaseMetaData.procedureColumnIn +
+        "                      THEN 'INPUT'" +
+        "                      WHEN V.COLUMN_TYPE = " + DatabaseMetaData.procedureColumnInOut +
+        "                      THEN 'INPUT_OUTPUT'" +
+        "                      WHEN V.COLUMN_TYPE = " + DatabaseMetaData.procedureColumnOut +
+        "                      THEN 'OUTPUT'" +
+        "                      ELSE NULL" +
+        "                  END" +
+        "            WHEN A.ALIASTYPE = 'F'" +
+        "            THEN CASE WHEN V.COLUMN_TYPE = " + DatabaseMetaData.functionColumnIn +
+        "                      THEN 'INPUT'" +
+        "                      WHEN V.COLUMN_TYPE = " + DatabaseMetaData.functionReturn +
+        "                      THEN 'RETURN_VALUE'" +
+        "                      ELSE NULL" +
+        "             END" +
+        "        END" +
+        "           AS COLUMN_DIRECTION" +
+        "     , V.DATA_TYPE" +
+        "           AS DATA_TYPE" +
+        "     , V.TYPE_NAME" +
+        "           AS TYPE_NAME" +
+        "     , CASE WHEN (V.COLUMN_TYPE = 5)" +
+        "            THEN CAST((V.PARAMETER_ID + 1 - V.METHOD_ID) AS INT)" +
+        "            ELSE CAST((V.PARAMETER_ID + 1) AS INT)" +
+        "        END" +
+        "           AS ORDINAL_POSITION" +
+        "     , V.PARAMETER_ID" +
+        "           AS PARAMETER_ID" +
+        "  FROM SYS.SYSALIASES A" +
+        "     , SYS.SYSSCHEMAS S" +
+        "     , NEW org.apache.derby.catalog.GetProcedureColumns(A.ALIASINFO, A.ALIASTYPE) V" +
+        " WHERE A.ALIASTYPE = '<<ProcFuncTypeFlag>>'" +
+        "   AND A.SCHEMAID = S.SCHEMAID" +
+        "   AND A.ALIAS = ?" +
+        "   AND S.SCHEMANAME = ?" +
+        " ORDER" +
+        "    BY PARAMETER_ID" +
+        "     , ORDINAL_POSITION";
+        String query = sql.replaceAll("<<ProcFuncTypeFlag>>", "P");
+        Map<String, DbParameterAccessor> params = fetchIntoParams(name, query);
+        if (!params.isEmpty()) {
+            return params;
```

Comment:
Let's imagine we have a `PROCEDURE` which doesn't have any parameters, and doesn't have a return value. What will happen in that case?